src/js/common/schemaform/state/reducers.js

```diff
@@ -12,6 +15,23 @@ import {
 } from '../state/helpers';
 
 export default {
+  [OPEN_REVIEW_CHAPTER]: (state, action) => {
+    const openChapters = _.concat(
+      _.get('reviewPageView.openChapters', state) || [],
+      action.openedChapter
+    );
+
+    return _.set('reviewPageView.openChapters', openChapters, state);
```

Comment:
Following the patterns from the other reducers but not mutating the state was engrained in me pretty early on in my redux training. Curious if there have been any re-rendering problems from doing this or if there's some other way this is mitigating problems that can happen from the simple object identity checking that `connect` does.

Reply:
Any lodash function you get from lodash/fp treats data as immutable, so we're not mutating the state here. Lodash does a shallow clone of the objects at each level in that path (and the root).

Reply:
Good to know- `object to modify` on the [lodash](https://gist.github.com/jfmengels/6b973b69c491375117dc#_setpath-value-object) docs let me to me think that it was.